scripts/ss_delete_bag.py

```diff
@@ -0,0 +1,459 @@
+    # invoke this script and delete a whole pile of bags at once.
+    _confirm_user_wants_to_delete_bag(
+        api_name=api_name,
+        space=space,
+        external_identifier=external_identifier,
+        version=version,
+        date_created=date_created,
+    )
+
+    reason = _ask_reason_for_deleting_bag(
+        space=space, external_identifier=external_identifier, version=version
+    )
+
+    # Do various other checks that this bag is correctly formed before we go
+    # ahead and start deleting stuff.  The deletion should be as close to atomic
+    # as possible -- we shouldn't get halfway through deleting the bag and then
+    # discover something is wrong; we should catch problems upfront.
+    storage_client = RequestsOAuthStorageServiceClient.from_path(api_url=api_url)
+
+    _confirm_is_latest_version_of_bag(
+        storage_client,
+        space=space,
+        external_identifier=external_identifier,
+        version=version,
+    )
+
+    bag = storage_client.get_bag(
+        space=space, external_identifier=external_identifier, version=version
+    )
+
+    # We only want to delete files that were newly introduced in this version --
+    # we shouldn't delete files fetch.txt'd from a prior version.
+    files_to_delete = [
+        f
+        for f in bag["manifest"]["files"] + bag["tagManifest"]["files"]
+        if f["path"].startswith(f"{version}/")
+    ]
+
+    locations = _confirm_user_wants_to_delete_locations(bag)
+    assert all(
+        loc["prefix"].endswith(f"/{version}")
+        for loc in locations["s3"] + [locations["azure"]]
+    )
+
+    environment = api_name
+    assert environment in ("prod", "staging")
+
+    # At this point, we've checked everything -- we're good to go!  Let's
+    # make a record of the deletion we're about to do.
+    dynamo_client = create_dynamo_client_from_role_arn(role_arn=DEV_ROLE_ARN)
+    _record_deletion(
+        dynamo_client,
+        environment=api_name,
+        ingest_id=ingest_id,
+        space=space,
+        external_identifier=external_identifier,
+        version=version,
+        reason=reason,
+    )
+
+    # And now start deleting stuff!
+    _delete_reporting_cluster_entries(
+        environment=environment,
+        ingest_id=ingest_id,
+        space=space,
+        external_identifier=external_identifier,
+        version=version,
+        s3_location=locations["s3"][0],
+        files_to_delete=files_to_delete,
+    )
+
+
+def hilight(s):
+    return click.style(str(s), "green")
+
+
+def abort(msg):
+    sys.exit(click.style(msg, "red"))
+
+
+def _confirm_user_wants_to_delete_bag(
+    api_name, space, external_identifier, version, date_created
+):
+    """
+    Show the user some information about the bag, and check this is really
+    the bag they want to delete.
+
+    It presents a prompt of the following form:
+
+          This is the bag you are about to delete:
+          Space:        testing
```

Comment:
Nit: I think this will also display the environment.

Reply:
Indeed it will; I realised I had to add that later.